Incorrect clone of argument closure in no-op implementation

The empty/no-op implementation of the `usdt` crate does not correctly clone the argument closure. As part of #35, we clone the argument closure passed to each probe. The closure is evaluated twice: in the context of argument type-checking and to generate the actual values to send to DTrace. (It's not actually evaluated in the first case, since that's wrapped in a closure that's never called, but the compiler doesn't know that.) This cloning is done in the empty implementation, but then the original closure is used in the second case again, rather than the local we store the closure into. I.e., it's transcribed twice, which leads to errors like this on Linux.

```bash
$ cargo +nightly check -p argument-types
    Checking argument-types v0.1.0 (/home/bnaecker/usdt/tests/argument-types)
error[E0382]: use of moved value: `arg`
   --> tests/argument-types/src/main.rs:21:1
    |
21  | #[usdt::provider]
    | ^^^^^^^^^^^^^^^^^ value used here after move
...
105 |     let arg = crate::Arg::default();
    |         --- move occurs because `arg` has type `Arg`, which does not implement the `Copy` trait
...
116 |     refs::serializable_as_reference!(|| arg);
    |     ----------------------------------------
    |     |                                |  |
    |     |                                |  variable moved due to use in closure
    |     |                                |  use occurs due to use in closure
    |     |                                value moved into closure here
    |     in this macro invocation
    |
    = note: this error originates in the macro `refs::serializable_as_reference` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0382`.
error: could not compile `argument-types` due to previous error
```

The actual problem can be seen using `cargo-expand`:

```bash
    let arg = crate::Arg::default();
    {
        let __usdt_private_args_lambda = || arg;
        #[allow(unused_imports)]
        fn __usdt_private_refs_serializable_as_reference_type_check(
            _: impl ::std::borrow::Borrow<crate::Arg>,
        ) {
        }
        let _ = || {
            let args = (__usdt_private_args_lambda.clone()(),);
            __usdt_private_refs_serializable_as_reference_type_check(args.0);
        };
        let _ = || (|| arg);
    };
```

That last `let` binding uses the literally-transcribed closure passed to the probe macro. It should use `__usdt_private_args_lambda`.

--- a/usdt-impl/src/empty.rs
+++ b/usdt-impl/src/empty.rs
@@ -50,7 +50,7 @@ fn compile_probe(
     probe: &Probe,
     config: &crate::CompileProvidersConfig,
 ) -> TokenStream {
-    let impl_block = quote! { let _ = || ($args_lambda); };
+    let impl_block = quote! { let _ = || (__usdt_private_args_lambda()) ; };
     common::build_probe_macro(
         config,
         provider,

